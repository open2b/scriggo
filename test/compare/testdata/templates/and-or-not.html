{# render #}

## AND

{% a1 := 20 %}{{ a1 and a1 }}
{% a2 := "something" %}{{ a2 and a2 }}
{% a4 := "" %}{{ a4 and a4 }}

## OR 

{% o1 := 20 %}{{ o1 or o1 }}
{% o2 := "something" %}{{ o2 or o2 }}
{% o4 := "" %}{{ o4 or o4 }}

## NOT

{% n1 := 4 %}{{ not n1 }}
{% n2 := 0 %}{{ not n2 }}
{% n3 := 0 %}{{ not not n3 }}

## MISC

{% m1 := 4 %}{% m2 := "hello" %}
{{ m1 and m2 }}
{{ m1 and not m2 }}
{{ not m1 and not m2 }}
{{ m1 and true or not m2}}

## INTERFACES

{{ true and interface{}(nil) }} nil
{{ true and interface{}(10) }} 10
{{ true and interface{}("") }} ""
{{ true and interface{}("hello") }} "hello"
{{ true and interface{}([]int{1, 2, 3}) }} []int{1, 2, 3}
{{ true and interface{}([]int{}) }} []int{}
{{ true and interface{}([]int(nil))}} []int(nil)

{{ interface{}(nil) or false }} nil
{{ interface{}(10) or false }} 10
{{ interface{}("") or false }} ""
{{ interface{}("hello") or false }} "hello"
{{ interface{}([]int{1, 2, 3}) or false }} []int{1, 2, 3}
{{ interface{}([]int{}) or false }} []int{}
{{ interface{}([]int(nil)) or false }} []int(nil)

{{ not not interface{}(nil) }} nil
{{ not not interface{}(10) }} 10
{{ not not interface{}("") }} ""
{{ not not interface{}("hello") }} "hello"
{{ not not interface{}([]int{1, 2, 3}) }} []int{1, 2, 3}
{{ not not interface{}([]int{}) }} []int{}
{{ not not interface{}([]int(nil)) }} []int(nil)