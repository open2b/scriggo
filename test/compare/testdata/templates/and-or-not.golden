## AND

true
true
false

## OR 

true
true
false

## NOT

false
true
false

## MISC


true
false
false
true

## INTERFACES

false nil
true 10
false ""
true "hello"
true []int{1, 2, 3}
false []int{}
false []int(nil)

false nil
true 10
false ""
true "hello"
true []int{1, 2, 3}
false []int{}
false []int(nil)

false nil
true 10
false ""
true "hello"
true []int{1, 2, 3}
false []int{}
false []int(nil)

## SLICES

[]int(nil):           false
[]int{}:              false
[]int{1, 2, 3}:       true
[]string{"", "", ""}: true